src/pages/api/v1/job-listings/[id].ts

```diff
@@ -0,0 +1,89 @@
+import prisma from '../../../../lib/prisma';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+const getJobListing = async (req: NextApiRequest, res: NextApiResponse) => {
+  try {
+    if (typeof req.query.id === 'string') {
+      const { id } = req.query;
+      const data = await prisma.jobListing.findUnique({
+        where: { id: parseInt(id) },
+      });
+      if (data) {
+        res.status(200).json(data);
+      } else {
+        res
+          .status(404)
+          .json({ message: `Could not find Job Listing with id ${id}` });
+      }
```

Comment:
Please rearrange the logic so the we early exit on an error, and have the success response last.